File: MetodosOrdenamiento/shell/Shell.java

import java.util.Random;

public class Shell {
    public static void shellSort(int[] arr) {
        int n = arr.length;
        
        for (int gap = n / 2; gap > 0; gap /= 2) {
            for (int i = gap; i < n; i++) {
                int temp = arr[i];
                int j;
                for (j = i; j >= gap && arr[j - gap] > temp; j -= gap) {
                    arr[j] = arr[j - gap];
                }
                arr[j] = temp;
            }
        }
    }
    
    public static void main(String[] args) {
        Random rand = new Random();
        int[] arr = new int[10];
        
        // Generar arreglo de 10 elementos con números aleatorios del 1 al 100
        for (int i = 0; i < 10; i++) {
            arr[i] = rand.nextInt(100) + 1;
        }
        
        System.out.print("Arreglo original: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
        
        shellSort(arr);
        
        System.out.print("Arreglo ordenado: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}

File: MetodosOrdenamiento/shell/shell.py
import random

def shell_sort(arr):
    n = len(arr)
    gap = n // 2
    
    while gap > 0:
        for i in range(gap, n):
            temp = arr[i]
            j = i
            while j >= gap and arr[j - gap] > temp:
                arr[j] = arr[j - gap]
                j -= gap
            arr[j] = temp
        gap //= 2

# Generar arreglo de 10 elementos con números aleatorios del 1 al 100
arr = [random.randint(1, 100) for _ in range(10)]
print("Arreglo original:", arr)

shell_sort(arr)
print("Arreglo ordenado:", arr)